### PR TITLE
React generator

### PR DESCRIPTION
**Pull request by Clarity-89**
**What this PR does / why we need it**:

A hackday project. A basic generator to scaffold React components and related files. 
So far supports creating:
- React functional components
- Test file for React components (test.tsx)
- Storybook story
- .mdx file with the props table for component

Could be easily expanded to support other use cases. 

generator

### CODE REVIEW

File: packages/grafana-toolkit/src/cli/index.ts [hunk 1]
```diff
@@ -219,6 +220,11 @@ export const run = (includeInternalScripts = false) => {
       await execTask(manifestTask)({ folder: process.cwd() });
     });
 
+  // React generator
+  program.command('component:create').action(async () => {
+    await execTask(componentCreateTask)({});
+  });
```
Comment (dprokop) [inline packages/grafana-toolkit/src/cli/index.ts:226]: Think this should be treated as internal task

File: packages/grafana-toolkit/src/cli/tasks/component.create.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    promptConfirm('hasTests', "Generate component's test file?"),
+    promptConfirm('hasStory', "Generate component's story file?"),
+    promptConfirm('isStoryPublic', 'Generate public story? (Selecting "No" will create an internal story)'),
+    promptInput(
+      'group',
+      'Select component group for the story (e.g. Forms, Layout)',
+      true,
+      'General',
+      ({ hasStory }) => hasStory
+    ),
```
Comment (dprokop) [inline packages/grafana-toolkit/src/cli/tasks/component.create.ts:36]: Would be nice if this was a select :)

**Review by dprokop — COMMENTED** [review]
This is looking cool. One concern - where is the component structure created when I run the task from i.e. root dir of grafana?

File: packages/grafana-toolkit/src/cli/tasks/component.create.ts (hunk 2 above)
Comment (Clarity-89) [reply packages/grafana-toolkit/src/cli/tasks/component.create.ts:36]: I was thinking the same but then you wanted to add a story for a new group, you'd have to manually modify the lost, altho not sure how often that would be used.

**Comment by Clarity-89** [comment]
> One concern - where is the component structure created when I run the task from i.e. root dir of grafana?

Good point, probably should mention it somewhere, the components will be created in the same folder the script is run from.

File: packages/grafana-toolkit/src/cli/index.ts (hunk 1 above)
Comment (Clarity-89) [reply packages/grafana-toolkit/src/cli/index.ts:226]: How can I make it an internal task?

Comment (dprokop) [reply packages/grafana-toolkit/src/cli/index.ts:232]: Put it within this condition: 
https://github.com/grafana/grafana/blob/e4b9e0cccd9e5fe335262555a19a4ba782e782cf/packages/grafana-toolkit/src/cli/index.ts#L24

**Review by dprokop — APPROVED** [review]
Cool! Can you make sure toolkit package builds before merging this? got to toolkit directory and run `yarn build`

**Pull request merged**